contrib/hyperscan/matching/input_matchers/source/matcher.cc

```diff
@@ -95,12 +59,88 @@ bool Matcher::match(absl::optional<absl::string_view> input) {
       },
       &matched);
   if (err != HS_SUCCESS && err != HS_SCAN_TERMINATED) {
-    ENVOY_LOG_MISC(error, "unable to scan, error code {}.", err);
+    ENVOY_LOG_MISC(error, "unable to scan, error code {}", err);
```

Comment:
should we include the failed regex string in this message?

Reply:
I am afraid we are not able to include the regex pattern in the message. Hyperscan cannot report the failed regex patterns on matching. In fact, the `else` block is just a insurance, the error can only be produced in the situation like lack memory or something else.

Reply:
I see -- I think if we wanted to retain the original regex we could, but given it's only an OOM situation it is not necessary.

Can you add a comment that we don't expect this to occur? Actually, should this be  an IS_ENVOY_BUG invocation rather than a log message?

We won't be able to cover this line in tests right?

Reply:
Thanks for your suggestion. We do not expect it to occur, and I will replace the log to a `IS_ENVOY_BUG`.

We will not cover these lines in tests.